Assets/Scripts/DeviceCommunicator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using UnityEngine;
+using Zenject;
+
+public class DeviceCommunicator : MonoBehaviour {
+
+	[Inject]
+	CodeExecutor codeExecutor;
+
+	public void Listen(string protobufInstructionsString) 
+	{
+		byte[] protobufInstructions = Encoding.Default.GetBytes (protobufInstructionsString);
+		Code code = Code.Parser.ParseFrom (protobufInstructions);
+
+		codeExecutor.Run (code);
+	}
+
+	[DllImport ("__Internal")]
+	private extern static void endLevel(LevelCompleteStatus status);
+
+	public void EndLevel(LevelCompleteStatus status) 
```

Comment:
so would the enum pass in as an int in the iOS code?

Reply:
Not sure, but rereading [this](https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types) makes me think that it's non-blittable so will need specific marshalling

Reply:
[Correction](https://stackoverflow.com/questions/22176159/how-to-marshal-a-c-enum-in-c-sharp)

Reply:
That works for C++ code so probs Objective-C code as well, but will it work for swift?

Reply:
/shrug. I just used the code that was linked originally. I'd have to do some research to see if it worked or not